index.js

```diff
@@ -762,6 +775,35 @@ class GithubScm extends Scm {
             sha: pullRequestInfo.head.sha
         }));
     }
+
+    /**
+     * Get an array of scm context (e.g. github.com)
+     * @method getScmContexts
+     * @return {Array}
+     */
+    _getScmContexts() {
+        const contextName = this.config.gheHost
+            ? [`github:${this.config.gheHost}`]
+            : ['github.com'];
+
+        return contextName;
+    }
+
+    /**
+     * Determine if a scm module can handle the received webhook
+     * @method canHandleWebhook
+     * @param {Object}    headers    The request headers associated with the webhook payload
+     * @param {Object}    payload    The webhook payload received from the SCM service
+     * @return {Promise}
+     */
+    _canHandleWebhook(headers, payload) {
+        return this._parseHook(headers, payload)
```

Comment:
I think I'm confused by the need for this method. Since we're going to call `scm.parseHook` again after calling it in the `chooseWebhookScm` method (in `canHandleWebhook`) anyways. Would it be more efficient to get rid of this method and put the `_canHandleWebhook` logic into the `scm-router` by calling `scm.parseHook` directly?
https://github.com/screwdriver-cd/scm-router/pull/1/files#diff-168726dbe96b3ce427e7fedce31bb0bcR170